arctic/store/_ndarray_store.py

```diff
@@ -36,6 +36,32 @@ def _promote(type1, type2):
     return np.dtype([(n, _promote(dtype1.fields[n][0], dtype2.fields.get(n, (None,))[0])) for n in dtype1.names])
 
 
+def _attempt_update_unchanged(symbol, unchanged_segment_ids, collection, version, previous_version):
+    if not unchanged_segment_ids or not collection or not version:
+        return
+    # Update the parent set of the unchanged/compressed segments
+    result = collection.update_many({
+                                        'symbol': symbol,  # hit only the right shard
```

Comment:
Could you verify that the `.explain(...` on a query of this shape is sane in the sharded context?  (I presume it will be...)

Reply:
Seems like it indeed hits only one shard

>>> expl = m['bloomberg.BLOOMBERG']._collection.find({'_id': ObjectId('55972d552914cf59a30ea0d5')}).explain()
>>> len(expl['queryPlanner']['winningPlan']['shards'])
24
>>> expl2 = m['bloomberg.BLOOMBERG']._collection.find({'symbol': 'BBG002RRJNS5', '_id': ObjectId('55972d552914cf59a30ea0d5')}).explain()
>>> len(expl2['queryPlanner']['winningPlan']['shards'])
1

Reply:
Also:

>>> expl2['queryPlanner']['winningPlan']['stage']
u'SINGLE_SHARD'

>>> expl['queryPlanner']['winningPlan']['stage']
u'SHARD_MERGE'